### PR TITLE
Create Apple Pay React Component

### PR DESCRIPTION
**Pull request by saltman-ck**

### CODE REVIEW

File: react-cardknox-ifields/package.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-  "version": "1.0.2",
+  "version": "1.1.0",
   "description": "A React component for Cardknox iFields",
   "license": "MIT",
   "main": "build/index.js",
   "dependencies": {
-    "prop-types": "^15.8.1",
-    "react": "^18.2.0"
+  "prop-types": "^15.8.1",
+  "react": "^18.2.0"
```
Comment (mstein-ck) [inline react-cardknox-ifields/package.json:9]: revert?

File: react-cardknox-ifields/src/lib.js
Comment (mstein-ck) [inline react-cardknox-ifields/src/lib.js:1]: Can this also depend on the `enableLogging` prop set on the component?

File: react-cardknox-ifields/src/services/httpService.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
+import { has, each, startsWith } from 'lodash';
```
Comment (mstein-ck) [inline react-cardknox-ifields/src/services/httpService.js:1]: `Lodash` needs to be added to the package

File: react-cardknox-ifields/src/services/httpService.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+import principalService from './principalService';
+import HttpServiceError from './httpServiceError';
+import { fixJsonFormat } from '../utilities/fix-json-format';
+import logger from '../utilities/logger';
+import updateApiEndpoint from 'common/utilities/UpdateApiEndpoint';
```
Comment (mstein-ck) [inline react-cardknox-ifields/src/services/httpService.js:7]: These imports don't seem to exist

File: react-cardknox-ifields/src/apple-pay.js [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+                                await this.validateMerchant();
+            const resp = typeof(response) === "string" ? JSON.parse(response) : response;
+            session.completeMerchantValidation(resp);
+        } catch (err) {
+            lib.logError("onValidateMerchant error", err);
+            session.abort();
+        }
+    }
+
+    validateMerchant = async () => {
```
Comment (mstein-ck) [inline react-cardknox-ifields/src/apple-pay.js:238]: Is the httpService here just for this call?
Are you able to use axios instead?
(Same for test app which already imports axios)

File: test/src/app.js [hunk 5]
```diff
@@ -1,4 +1,183 @@
...
+                console.log('applePayPaymentAuthorize', paymentResponse);
+                resolve(paymentResponse);
+            } catch(error) {
+                lib.logError("onPaymentAuthorize error.", error);
+                reject(error);
+            }
+        });
+    }
+}
 import AppForm from './Components/form';
```
Comment (mstein-ck) [inline test/src/app.js:181]: All import statements should be at the top of the file

File: test/src/app.js [hunk 6]
```diff
@@ -1,4 +1,183 @@
+import React from 'react';
+// import IField, { CARD_TYPE, CVV_TYPE } from '@cardknox/react-ifields';
+import CardknoxApplePay from './apple-pay';
+import * as lib from './lib';
 
+export default class App extends React.Component {
```
Comment (mstein-ck) [inline test/src/app.js:6]: You can't have multiple default exports

File: test/src/app.js [hunk 7]
```diff
@@ -1,4 +1,183 @@
+import React from 'react';
+// import IField, { CARD_TYPE, CVV_TYPE } from '@cardknox/react-ifields';
```
Comment (mstein-ck) [inline test/src/app.js:2]: Should this be removed?

File: test/src/app.js [hunk 8]
```diff
@@ -1,4 +1,183 @@
...
+            merchantIdentifier: 'merchant.aptest.cardknoxdev.com',
+            requiredShippingContactFields: ['postalAddress']
+        }
+    }
+
+    render() {
+        return (
+            <>
+
+                {/* <IField
```
Comment (mstein-ck) [inline test/src/app.js:70]: Should this be removed?
Also, where is the sample for regular ifields?

File: react-cardknox-ifields/src/apple-pay.js
Comment (mstein-ck) [inline react-cardknox-ifields/src/apple-pay.js:1]: This just seems like a copy of the file in the test app. Which one is needed?

File: test/src/styles/main.css
Comment (mstein-ck) [inline test/src/styles/main.css:1]: Seems to be duplicate of the file in the actual component

File: react-cardknox-ifields/src/apple-pay.js
Comment (saltman-ck) [reply react-cardknox-ifields/src/apple-pay.js:1]: Both.
The testing version is in test and release version is in component

File: test/src/styles/main.css
Comment (saltman-ck) [reply test/src/styles/main.css:1]: Same reason as above

File: react-cardknox-ifields/src/lib.js
Comment (saltman-ck) [reply react-cardknox-ifields/src/lib.js:1]: Done

File: react-cardknox-ifields/package.json (hunk 1 above)
Comment (saltman-ck) [reply react-cardknox-ifields/package.json:9]: Done

File: react-cardknox-ifields/src/services/httpService.js (hunk 3 above)
Comment (saltman-ck) [reply react-cardknox-ifields/src/services/httpService.js:7]: Cleaned up

File: react-cardknox-ifields/src/apple-pay.js (hunk 4 above)
Comment (saltman-ck) [reply react-cardknox-ifields/src/apple-pay.js:238]: Discussed

File: test/src/app.js (hunk 6 above)
Comment (saltman-ck) [reply test/src/app.js:6]: Resolved

File: test/src/app.js (hunk 7 above)
Comment (saltman-ck) [reply test/src/app.js:2]: Discussed

File: test/src/app.js (hunk 8 above)
Comment (saltman-ck) [reply test/src/app.js:70]: Discussed

File: react-cardknox-ifields/index.d.ts [hunk 9]
```diff
@@ -1,3 +1,4 @@
+import { string } from "prop-types"
```
Comment (mstein-ck) [inline react-cardknox-ifields/index.d.ts:1]: Is this needed? It looks like 'string' was already being used

File: react-cardknox-ifields/package.json [hunk 10]
```diff
@@ -1,13 +1,9 @@
 {
   "name": "@cardknox/react-ifields",
-  "version": "1.0.2",
+  "version": "1.2.60",
```
Comment (mstein-ck) [inline react-cardknox-ifields/package.json:3]:
```suggestion
  "version": "2.2.60",
```

Comment (mstein-ck) [reply react-cardknox-ifields/package.json:3]: Need to rerun `npm install`

File: react-cardknox-ifields/package.json [hunk 11]
```diff
@@ -1,13 +1,9 @@
...
   "description": "A React component for Cardknox iFields",
   "license": "MIT",
   "main": "build/index.js",
-  "dependencies": {
-    "prop-types": "^15.8.1",
-    "react": "^18.2.0"
```
Comment (mstein-ck) [inline react-cardknox-ifields/package.json:9]: I believe 'react' is needed (if you don't have react installed globally)

File: test/.gitignore
Comment (mstein-ck) [inline test/.gitignore:1]: Are you using amplify?
Can this be moved to the gitignore at root level?

File: test/src/app.js [hunk 12]
```diff
@@ -1,9 +1,179 @@
...
+            autoFormatSeparator: ' ',
+            autoSubmit: false,
+            iFieldstyle: {
 
-export default function App() {
+            },
+            iFrameStyle: {
 
-    return (
-        <AppForm />
```
Comment (mstein-ck) [inline test/src/app.js:7]: Can the apple pay component be added to the `form` component?
Removing that component removes all the 3DS sample code

File: react-cardknox-ifields/src/index.js [hunk 13]
```diff
@@ -1,9 +1,11 @@
...
 import IField from "./iField";
 export default IField;
+export {default as CardknoxApplePay } from "./apple-pay";
+
 export {
     CARD_TYPE,
     CVV_TYPE,
     ACH_TYPE,
-    THREEDS_ENVIRONMENT
+    THREEDS_ENVIRONMENT,
```
Comment (mstein-ck) [inline react-cardknox-ifields/src/index.js:10]: If you are exporting the apple pay constants for consumer to use, you need to export them here.
If not, they can be removed from the index.d.ts file

File: test/.gitignore
Comment (saltman-ck) [reply test/.gitignore:1]: I'm not using amplify

File: react-cardknox-ifields/index.d.ts (hunk 9 above)
Comment (saltman-ck) [reply react-cardknox-ifields/index.d.ts:1]: Removed

File: react-cardknox-ifields/package.json (hunk 10 above)
Comment (saltman-ck) [reply react-cardknox-ifields/package.json:3]: done

File: react-cardknox-ifields/package.json (hunk 11 above)
Comment (saltman-ck) [reply react-cardknox-ifields/package.json:9]: Put it back (you told me to revert it...)

File: test/src/app.js (hunk 12 above)
Comment (saltman-ck) [reply test/src/app.js]: Done

File: react-cardknox-ifields/src/index.js (hunk 13 above)
Comment (saltman-ck) [reply react-cardknox-ifields/src/index.js:10]: Discussed

**Review by mstein-ck — APPROVED** [review]

**Pull request merged**